server/android_notification_server.go

```diff
@@ -220,3 +213,31 @@ func (me *AndroidNotificationServer) SendNotification(msg *PushNotification) Pus
 	}
 	return NewOkPushResponse()
 }
+
+func isRetriableError(err error) bool {
+	// Any error that is not a known error, but the Internal error
+	return !(messaging.IsUnregistered(err) ||
+		messaging.IsInvalidArgument(err) ||
+		messaging.IsQuotaExceeded(err) ||
+		messaging.IsSenderIDMismatch(err) ||
+		messaging.IsThirdPartyAuthError(err))
+}
+
+func (me *AndroidNotificationServer) SendNotificationWithRetry(fcmMsg *messaging.Message, retry int) error {
+	ctx, cancel := context.WithTimeout(context.Background(), me.sendTimeout)
+	defer cancel()
+
+	start := time.Now()
+	_, err := me.client.Send(ctx, fcmMsg)
+	if me.metrics != nil {
+		me.metrics.observerNotificationResponse(PushNotifyAndroid, time.Since(start).Seconds())
+	}
+
+	if isRetriableError(err) {
+		if nextIteration := retry + 1; nextIteration < MAX_RETRIES {
+			return me.SendNotificationWithRetry(fcmMsg, nextIteration)
+		}
```

Comment:
Should we add a metric or log showing that the max retries was reached?

Reply:
Yes, we probably should.